src/services/exit_order.py

```diff
@@ -1,24 +1,285 @@
+    3. Validator whose operator with the highest stake weight
+    4. Validator whose operator with the highest number of validators
+    5. Validator with the lowest activation epoch
+    6. Validator with the lowest index
     """
-    def __init__(self, blockstamp: BlockStamp, w3: Web3,  max_size: int):
+
+    def __init__(
+        self,
+        blockstamp: BlockStamp,
+        w3: Web3,
+        c_conf: ChainConfig,
+        exitable_lido_validators: list[LidoValidator],
+        max_validators_to_exit: int,
+    ):
         self.blockstamp = blockstamp
         self.w3 = w3
+        self.c_conf = c_conf
+
+        self.exitable_lido_validators = exitable_lido_validators
+        self.max_validators_to_exit = max_validators_to_exit
 
-        # Iterator should contain only next validators
-        self.max_size = max_size
+        ###########
+        # todo: remove after keys-api fix. Key should contain staking module id, not only address
+        operators = self.w3.lido_validators.get_lido_node_operators(blockstamp)
+        self.staking_module_id = {
+            operator.staking_module.staking_module_address: operator.staking_module.id
+            for operator in operators
+        }
+        self.no_index_by_validator = (
+            lambda v: NodeOperatorIndex(self.staking_module_id[v.key.moduleAddress], v.key.operatorIndex)
+        )
+        ###########
+        operator_validators = self.w3.lido_validators.get_lido_validators_by_node_operators(blockstamp)
+        self.lido_node_operator_stats = self._prepare_lido_node_operator_stats(
+            blockstamp, operators, operator_validators
+        )
+        self.total_predictable_validators = sum(
+            operator_state.predictable_validators_count for operator_state in self.lido_node_operator_stats.values()
+        )
 
     def __iter__(self):
         return self
 
-    def __next__(self):
+    def __next__(self) -> LidoValidator:
+        left_queue_count = 0
+        while left_queue_count < self.max_validators_to_exit:
+            self.exitable_lido_validators.sort(key=lambda validator: self._predicates(validator))
+            to_exit = self.exitable_lido_validators.pop(0)
+            yield to_exit
+            self._decrease_node_operator_stats(to_exit)
+            left_queue_count += 1
         raise StopIteration
+
+    def _decrease_node_operator_stats(self, validator: LidoValidator) -> None:
+        """
+        Sub particular validator stats from its node operator stats
+        We do it every time when validator is popped from the queue for resort the rest of queue
+        """
+        module_operator = self.no_index_by_validator(validator)
+        self.total_predictable_validators -= 1
+        self.lido_node_operator_stats[module_operator].predictable_validators_count -= 1
+        self.lido_node_operator_stats[module_operator].predictable_validators_total_age -= int(validator.validator.activation_epoch)
```

Comment:
Not sure, but likely it would be `blockstamp.ref_epoch - int(validator.validator.activation_epoch)` as from total_age summarise